sunspot/lib/sunspot/data_extractor.rb

```diff
@@ -53,18 +53,23 @@ def initialize(value)
       end
 
       def value
-        strip_control_characters @value
-      end
-
-      def strip_control_characters(value)
-        if value.is_a? String
-          value.gsub(/[[:cntrl:]]/, ' ')
-        elsif value.is_a? Array
-          value.map { |v| strip_control_characters(v) }
+        if @value.is_a? String
+          strip_control_characters_from_string @value
+        elsif @value.is_a? Array
+          @value.map { |v| strip_control_characters_from_string v }
+        elsif @value.is_a? Hash
+          @value.inject({}) do |hash, (k, v)|
+            hash.merge(strip_control_characters_from_string(k) => strip_control_characters_from_string(v))
+          end
         else
-          value
+          @value
         end
       end
+
+      def strip_control_characters_from_string(value)
+        return value unless value.is_a? String
+        value.gsub /[^[:print:]\r\n]/, ''
```

Comment:
maybe replace it with a space rather than removing it? For cases like, "hello\nworld" => "hello world" instead of "helloworld"